types/mongoose/index.d.ts

```diff
@@ -2021,6 +2021,30 @@ declare module "mongoose" {
      */
     or(array: any[]): this;
 
+    /**
+     * Make this query throw an error if no documents match the given `filter`.
+     * This is handy for integrating with async/await, because `orFail()` saves you
+     * an extra `if` statement to check if no document was found.
+     *
+     * ####Example:
+     *
+     *     // Throws if no doc returned
+     *     await Model.findOne({ foo: 'bar' }).orFail();
+     *
+     *     // Throws if no document was updated
+     *     await Model.updateOne({ foo: 'bar' }, { name: 'test' }).orFail();
+     *
+     *     // Throws "No docs found!" error if no docs match `{ foo: 'bar' }`
+     *     await Model.find({ foo: 'bar' }).orFail(new Error('No docs found!'));
+     *
+     *     // Throws "Not found" error if no document was found
+     *     await Model.findOneAndUpdate({ foo: 'bar' }, { name: 'test' }).
+     *       orFail(() => Error('Not found'));
+     *
+     * @param err optional error to throw if no docs match `filter`
+     */
+    orFail(err?: any): Query<any> & QueryHelpers;
```

Comment:
The parameter should be a function or error object.
```suggestion
    orFail(err?: Error | (() => Error)): Query<any> & QueryHelpers;
```